### PR TITLE
deprecated(core): renamed various session related lib functions

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: engine/classes/ElggSession.php [hunk 1]
```diff
@@ -244,6 +246,90 @@ public function remove($name) {
 	public function has($name) {
 		return $this->storage->has($name);
 	}
+	
+	/**
+	 * Logs in a specified \ElggUser. For standard registration, use in conjunction
```
Comment (jeabakker) [inline engine/classes/ElggSession.php:251]: Please have a look at this description, I find it confusing

File: engine/classes/ElggSession.php [hunk 2]
```diff
@@ -244,6 +246,90 @@ public function remove($name) {
...
+		}
+	
+		// give plugins a chance to reject the login of this user (no user in session!)
+		if (!elgg_trigger_before_event('login', 'user', $user)) {
+			throw new LoginException(elgg_echo('LoginException:Unknown'));
+		}
+	
+		if (!$user->isEnabled()) {
+			throw new LoginException(elgg_echo('LoginException:DisabledUser'));
+		}
```
Comment (jeabakker) [inline engine/classes/ElggSession.php:275]: shouldn't this be the first check

File: engine/classes/ElggSession.php [hunk 3]
```diff
@@ -244,6 +246,90 @@ public function remove($name) {
...
+	
+		if (!elgg_trigger_before_event('logout', 'user', $user)) {
+			return false;
+		}
+	
+		_elgg_services()->persistentLogin->removePersistentLogin();
+	
+		// pass along any messages into new session
+		$old_msg = $this->get(SystemMessagesService::SESSION_KEY, []);
+		$this->invalidate();
```
Comment (jeabakker) [inline engine/classes/ElggSession.php:326]: this removes the user from the session?

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: engine/classes/ElggSession.php (hunk 2 above)
Comment (jdalsem) [reply engine/classes/ElggSession.php:275]: agree

File: engine/classes/ElggSession.php (hunk 3 above)
Comment (jdalsem) [reply engine/classes/ElggSession.php:326]: yep

File: engine/classes/ElggSession.php (hunk 1 above)
Comment (jdalsem) [reply engine/classes/ElggSession.php:251]: me 2

**Review by jeabakker — APPROVED** [review]

**Pull request merged**